### PR TITLE
allow patterns in text and invalid input messages

### PR DESCRIPTION
**Pull request by wizehive-chrismanjoine**

### CODE REVIEW

File: src/inputs/text.html [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
 	<input type="text" class="form-control" name="{{ field.id }}"
-		ng-model="editing.config[field.id]"
-		ng-required="field.required"
-		placeholder="{{ field.placeholder }}"
-		ng-minlength="{{ field.minlength }}"
-		ng-maxlength="{{ field.maxlength }}">
+		   ng-model="editing.config[field.id]"
+		   ng-required="field.required"
+		   placeholder="{{ field.placeholder }}"
+		   ng-pattern="field.pattern"
```
Comment (jscarton) [inline src/inputs/text.html:7]: I believe this should be `{{ field.required }}` and `{{ field.pattern }}`?

Comment (wizehive-chrismanjoine) [reply src/inputs/text.html:7]: it actually isn't because they are not strings the field.pattern is a regex and field.required is a boolean

Comment (wizehive-chrismanjoine) [reply src/inputs/text.html:7]: the field.pattern doesn't work with {{}} took me for ever to realize that :(

Comment (jscarton) [reply src/inputs/text.html:7]: got it, thanks for clarification

**Review by jscarton — APPROVED** [review]
LGTM!

**Pull request merged**